whylog/constraints/__init__.py

```diff
@@ -153,24 +177,25 @@ class TimeConstraint(AbstractConstraint):
     MIN_DELTA = 'min_delta'
     MAX_DELTA = 'max_delta'
 
-    PARAMS = sorted([MIN_DELTA, MAX_DELTA])
+    PARAMS = {MIN_DELTA: ConverterType.TO_FLOAT, MAX_DELTA: ConverterType.TO_FLOAT}
 
     def __init__(self, groups=None, param_dict=None, params_checking=True):
         super(TimeConstraint, self).__init__(groups, param_dict, params_checking)
-        param_min_delta = self.params.get(self.MIN_DELTA)
-        param_max_delta = self.params.get(self.MAX_DELTA)
-        if param_min_delta is not None and param_max_delta is not None:
-            self.verify = self._verify_both
-            self._min_delta = datetime.timedelta(seconds=param_min_delta)
-            self._max_delta = datetime.timedelta(seconds=param_max_delta)
-        elif param_max_delta is not None:
-            self.verify = self._verify_max
-            self._max_delta = datetime.timedelta(seconds=param_max_delta)
-        elif param_min_delta is not None:
-            self.verify = self._verify_min
-            self._min_delta = datetime.timedelta(seconds=param_min_delta)
-        else:
-            raise WrongConstraintClassSetup(self.TYPE)
+        if not params_checking:
```

Comment:
If params_checking is False this if will return True and you will check param. I guess you want to write here `if param_checking:`.